### DISCUSSION TITLE
Add the ability to move the table of contents to the right side of the page

### DISCUSSION THREAD

**Discussion by ghost** [outline/outline#6037] [Ideas]
Currently the table of contents for pages can either be shown or hidden. When it is shown it appears on the left side of the page (between the page content and the navigation side bar).

Please add the ability to toggle between left side or right side of the page for the table of contents.

Ideally this would be a global default option, with the ability for admins to allow this setting to be user specific (or not)

Current state:
![image](https://github.com/outline/outline/assets/77940332/708ba3d7-b36e-4cc9-8bbe-be556356f94d)

Proposed option:
![image](https://github.com/outline/outline/assets/77940332/c2598f96-347e-4814-8cc2-efa2979bbd5c)

**Comment by ReessKennedy** [comment]
+1 ... I view the right side as the more common side for a table of contents. Consider that Docusarus and almost all of the very popular static builders have the TOC on the right. Clickup's is to the right and WikiJS provides and option for left or right. Attached is just one I could find of dozens on different static builders ... and this all conditions people to increasingly, I think, look for the TOC on the right: 

<img width="1491" alt="Screen Shot 2023-12-07 at 10 28 10 PM" src="https://github.com/outline/outline/assets/3970267/ee3eb6b8-38f2-4847-96d0-01ac69b3b1e3">

**Reply by ReessKennedy** [reply]
... reviewing this a few months later. Was hoping these examples would give solid support for this as a valuable option/improvement to bring Outline into harmony with what seems to be the default docs layout instead of folks being locked into an alternate layout. @tommoor -- continued congrats on this tool.

**Reply by tommoor** [reply]
The plan is still to address this with a preference, the issue is the shortage of time not a shortage of examples

**Reply by ReessKennedy** [reply]
Awesome! I get it! Thanks for all your work @tommoor

**Comment by EduNetEurope** [comment]
Duplicate of https://github.com/outline/outline/discussions/3214

**Comment by ReessKennedy** [comment]
There's actually a simple way to do this. ✅ 🎉 ... not sure why I didn't figure this out sooner. Wrote up solution at link below but's basically just a CSS mod with optional JS addition to make the table of contents show up by default: 

[https://www.reesskennedy.com/right-sidebar-outline-wiki/](https://www.reesskennedy.com/right-sidebar-outline-wiki/)

![BeforeAndAft](https://github.com/outline/outline/assets/3970267/4a92b4bc-0db5-4d2b-a3c4-b52ebbc3aaa3)

**Comment by EduNetEurope** [comment]
@tommoor 
If it is that simple, why don't you add the option to move the ToC to the right side?

**Reply by ReessKennedy** [reply]
In fairness to @tommoor there are likely things I am overlooking but my override seems to work! Flexbox makes rearrangement like this "pretty, pretty good."

**Reply by tommoor** [reply]
Yea, it breaks down pretty quickly at wide sizes – but if there is such appetite I can prioritize this change

<img width="946" alt="image" src="https://github.com/outline/outline/assets/380914/41abce4e-ff70-41ff-b313-3780e928e0f9">

**Reply by ReessKennedy** [reply]
I just updated my CSS override to handle normal site widths as well! My original just did full width. Still could use some refining and still a hack but feeling good about it: 

```css
/* Full width **/
.eSMets {
    flex-direction: row !important; 

}

/* Normal width **/
.dyKbFK {
    display: block !important;
}	

@media (min-width: 100em) {
    .lcznko {
        transform: translateX(-5px) !important;
    }
}
```

**Reply by ReessKennedy** [reply]
And yeah, I think there is appetite -- 3 or 4 different threads on this. You the man @tommoor!

**Comment by ghost** [comment]
I gave up on outline. Simple stuff like this that has been requested for 2+ years was missing and we went with another product.

**Comment by ReessKennedy** [comment]
Looks like this was added as a core feature! I'd been carrying on using my CSS track that was working well but now no longer needed. NIce!

**Reply by ghost** [reply]
> Looks like this was added as a core feature! I'd been carrying on using my CSS track that was working well but now no longer needed. NIce!

where are you seeing this has been added as a core feature? I just stood up a new trial and still don't see any options or preferences to move the TOC to the right hand side.

**Reply by EduNetEurope** [reply]
Best news of the day!!! Yes, it was. THANK YOU, @tommoor. Amazing!!!

![image](https://github.com/user-attachments/assets/a4487c18-1ef6-4675-b594-b6f90f460b29)
-> Preferences
-> Workspace / Details: Table of contents position.

There are a number of requests that can be closed now.

**Comment by EduNetEurope** [comment]
@tommoor 
I think you can close this request. :-)
